services/masterdata/src/lib/core/services/masterdata-renderer.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from "@angular/core";
+import { TranslateService } from "@ngx-translate/core";
+
+type Entries<T> = {
+	[K in keyof T]: [K, T[K]];
+}[keyof T][];
+
+@Injectable()
+export class MasterdataRendererService {
+	constructor(
+		private readonly translateService: TranslateService
+	) {}
+
+	prepareForDisplay<T extends object>(item: T, translationKey: string): Entries<T> {
+		// "metadata" model field will have a special display!
+		return (Object.entries(item) as Entries<T>).filter(item => item[0] !== "metadata").map(item => {
+			item[0] = this.translateService.instant(`masterdatamgmt.pages.${translationKey}.modelFields.${item[0].toString()}`);
+			return item;
```

Comment:
Why is the duplicate return statement still not removed?

Reply:
@Johanh-od it gives me a typescript error if i don't return something!

Reply:
The ln 18 return is return in the map-callback function. The return on line 16 returns the result of the call to map. So this is correct.

Reply:
ah, now I see. Its a bit difficult to read due to the formatting

Reply:
it was as you said... a code smell! trying to chain the javascript methods, basically generated the issue!

Reply:
@Johanh-od changed the map with forEach as in my code case, forEach si more appropiate (I don't want to return a new array as with map).